### PR TITLE
PUBDEV-7304: Add GBM POJO support for SortByResponse and EnumLimited encodings

### PR DESCRIPTION
**Pull request by koniecsveta**
https://0xdata.atlassian.net/browse/PUBDEV-7304

### CODE REVIEW

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class EnumLimitedEncoder implements CategoricalEncoder {
+
+  private final String columnName;
+  private final int targetIndex;
+  private final Map<String, Integer> domainMap;
+  
+  EnumLimitedEncoder(String columnName, int targetIndex, String[] domainValues, String[] newDomainValues) {
+    this.columnName = columnName;
+    this.targetIndex = targetIndex;
+    domainMap = new HashMap<>(domainValues.length);
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java]: can be moved to Line#10

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package hex.genmodel.easy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EnumLimitedEncoder implements CategoricalEncoder {
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java]: this class is a great candidate for a unit test

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoderColumnMapper.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package hex.genmodel.easy;
+
+import hex.genmodel.GenModel;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EnumLimitedEncoderColumnMapper {
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoderColumnMapper.java]: this looks exactly the same as EnumEncoderColumnMapper we should not just copy-paste code

Comment (koniecsveta) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoderColumnMapper.java]: there is a difference in line 17: EnumLimitedEncoderColumnMapper takes _m.getOrigNames() as modelColumnNames and EnumEncoderColumnMapper takes _m.getNames() as modelColumnNames

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java (hunk 2 above)
Comment (koniecsveta) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EnumLimitedEncoder.java]: added tests for all new classes

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EnumEncoderColumnMapper.java [hunk 4]
```diff
@@ -7,14 +7,19 @@
 
 public class EnumEncoderColumnMapper {
 
-  private final GenModel _m;
+  final GenModel _m;
+  String[] modelColumnNames;
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EnumEncoderColumnMapper.java]: there is no reason to make this a instance variable a simple protected getModelColumnNames() method would be enough, together with making _m protected

Comment (koniecsveta) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EnumEncoderColumnMapper.java]: oh i see, fixed

Comment (honzasterba) [reply h2o-genmodel/src/main/java/hex/genmodel/easy/EnumEncoderColumnMapper.java]: 👍

**Pull request merged**